### PR TITLE
WIP check root group attr names, not running tests

### PR DESCRIPTION
**Pull request by benjwadams**

### CODE REVIEW

**Comment by benjwadams** [comment]
`py.test` does not seem to be running tests for the added unit tests on my machine, not sure why -- it is inside `TestCF1_8` class and has the name `test_groups`, which should be picked up by py.test to my understanding, or have I missed something?

**Comment by benjwadams** [comment]
Weird, codecov isn't reflecting the test addition either and should also be running py.test.

File: compliance_checker/tests/test_cf.py [hunk 1]
```diff
@@ -2378,6 +2378,23 @@ class TestCF1_8(BaseTestCase):
...
 
+    def test_groups(self):
+        dataset = MockTimeSeries()
+        # TEST CONFORMANCE 2.7 REQUIRED 1/4
+        nonroot_group = dataset.createGroup("nonroot")
+        dummy = nonroot_group.createVariable("dummy")
+        dummy.Conventions = "CF-1.8"
+        dummy.external_variables = "ext1"
+        results = self.cf.check_groups(dataset)
+        bad_msg_template = ("§2.7.2 Note: attribute '{}' found on non-root "
```
Comment (ocefpaf) [inline compliance_checker/tests/test_cf.py:2389]:
```suggestion
        bad_msg_template = ("§2.7.2 Note: attribute '{attr_name}' found on non-root "
```

File: compliance_checker/tests/test_cf.py [hunk 2]
```diff
@@ -2378,6 +2378,23 @@ class TestCF1_8(BaseTestCase):
...
+                            "group 'nonroot'. It is allowed in order to provide "
+                            "additional provenance and description of the "
+                            "subsidiary data. It does not override "
+                            "attributes from the parent groups.")
+        bad_messages = {bad_msg_template.format(attr_name) for attr_name in
```
Comment (ocefpaf) [inline compliance_checker/tests/test_cf.py:2394]:
```suggestion
        bad_messages = {bad_msg_template.format(attr_name=attr_name) for attr_name in
```

**Comment by benjwadams** [comment]
Closed and reopened to run tests against current `develop` branch.

**Pull request merged**